### ISSUE TITLE
amp-<test> segfaults with unknown options

### ISSUE THREAD

**Issue by zafnz** [wandnz/amplet2#3]
When specifying an unknown/invalid option, various amp tests (eg amp-icmp) segfaults.

Expected result:
```
root@tut5:~# amp-icmp --i-am-a-bad-user
Usage: amp-icmp [-hrvx] [-p perturbate] [-s packetsize]
                [-Q codepoint] [-Z interpacketgap]...<snip>
```

Actual result:
```
root@tut5:~# amp-icmp --i-am-a-bad-user
Segmentation fault (core dumped)
```

Note, this also happens with amp-icmp --- (3 dashes), but this is probably from the same bug (you aren't handling the error result from getopt_long correctly.

**Comment by zafnz** [comment]
So the pull request closes the above. I don't know if you prefer the issues closed immediately, or only after the pull request is complete.

**Comment by brendonj** [comment]
Merged #11. Fixes the issue for all tests that share the code from testmain.c. Thanks!
